File: c++/cuentaAtras.cc
/* ----------------------------------------------------------------------------
 * File:   cuentaAtras.cc
 * Author: Pablo Angusto Delgado 842255 y Miguel Arejulajula Aisa 850068
 * Date:   marzo 2023
 * Coms:   Practica 2 de Tecpro 
 *         
 * -----------------------------------------------------------------------------
 */
#include "cuentaAtras.h"


CuentaAtras::CuentaAtras() : Programa(){
            fin=7;
            vInst = new Instruccion*[fin];
            vInst[0]= new Read();
            vInst[1] = new Dup();
            vInst[2] = new Write();
            vInst[3] = new Push(-1);
            vInst[4] = new Add();
            vInst[5] = new Dup();
            vInst[6] = new Jumpif(1);
            cont=0;
}

CuentaAtras::~CuentaAtras(){
   for (int i = 0; i < fin; i++)
    {
        delete vInst[i];
    }
    
}

    

File: java/CuentaAtras.java
/* ----------------------------------------------------------------------------
 * File:   CuentaAtras.java
 * Author: Pablo Angusto Delgado 842255 y Miguel Aréjula Aisa 850068
 * Date:   marzo 2023
 * Coms:   Practica 2 de Tecpro 
 *         
 * -----------------------------------------------------------------------------
 */
 public class CuentaAtras extends Programa {
    public CuentaAtras(){
        info.cont = 0;
        info.fin = 7;
        vInst = new Instruccion[info.fin];
        vInst[0] = new Read()
        vInst[0] = new Dup()
        vInst[0] = new Write()
        vInst[0] = new Push(-1)
        vInst[0] = new Add()
        vInst[0] = new Dup()
        vInst[0] = new Jumpif(1)
    }
    
}
